### ISSUE TITLE
Potential read_dipole API bug

### ISSUE THREAD

**Issue by chenghuzi** [jonescompneurolab/hnn-core#617]
I believe the functionality of this API:
https://github.com/jonescompneurolab/hnn-core/blob/d5d1a03e98911670a9721cb674331e5281df7e31/hnn_core/dipole.py#L118

is the same as the one we previously used here: 

https://github.com/jonescompneurolab/hnn/blob/49e553c0d3b9ac8b7f64f01e72e90f690525f89f/hnn/qt_main.py#L242-L252

However, it seems that the new API does not have the same level of tolerance for the data format. Should we add this functionality, or can we safely ignore it since all new data is now stored in a compatible way?

**Comment by jasmainak** [comment]
I think we should definitely add this functionality. The main question is what should the data structure in the Dipole object be like? Currently `dpl.data` is a dictionary of arrays with keys 'L2', 'L5', and 'agg'. Do we use the key 'exp' to indicate experimental data? Any other suggestions @rythorpe @ntolley @chenghuzi ?

**Comment by jasmainak** [comment]
For multiple trials, we should return a list of dipoles ... I think that's how we currently treat multiple trials during simulation

**Comment by rythorpe** [comment]
I think the heart of this issue is the same as #520.

**Comment by rythorpe** [comment]
The previous I/O convention for HNN was that each dipole trial gets mapped to 1 and only 1 .txt file. The `S1_ongoing.txt` experimental data file breaks this convention, which I'm guessing is the real culprit here. Is this correct @chenghuzi? If so, we should either fix `S1_ongoing.txt` or establish a new convention, but we should be consistent throughout our `load_dipole()` and `Dipole.write()` functions.

**Comment by jasmainak** [comment]
I absolutely agree ... so the convention was to have one `txt` for one dipole file, right? I'm fine either way ... just establish one and stick to it throughout.

**Comment by chenghuzi** [comment]
> For multiple trials, we should return a list of dipoles ... I think that's how we currently treat multiple trials during simulation

That's also the way we handle uploaded `txt`s in the PR

**Comment by chenghuzi** [comment]
> The previous I/O convention for HNN was that each dipole trial gets mapped to 1 and only 1 .txt file. The `S1_ongoing.txt` experimental data file breaks this convention, which I'm guessing is the real culprit here. Is this correct @chenghuzi? If so, we should either fix `S1_ongoing.txt` or establish a new convention, but we should be consistent throughout our `load_dipole()` and `Dipole.write()` functions.

Besides the `S1_ongoing.txt` there is also [100_trials.txt](https://github.com/jonescompneurolab/hnn/blob/master/data/gamma_tutorial/100_trials.txt).

**Comment by rythorpe** [comment]
> Besides the `S1_ongoing.txt` there is also [100_trials.txt](https://github.com/jonescompneurolab/hnn/blob/master/data/gamma_tutorial/100_trials.txt).
> 
> Personally I prefer just allowing the `read_dipole` to accept single trials

I agree (at least until we can revamp our I/O routines), except that this still leaves us with the issue you brought up [here](https://github.com/jonescompneurolab/hnn-core/pull/616#discussion_r1136382414). The ideal situation is where the GUI code acts as a wrapper for the HNN-core API, maintaining all HNN-core I/O conventions.

**Comment by rythorpe** [comment]
This leaves us with one of two options: either we separate `S1_ongoing.txt` and `100_trials.txt` into individual trials for their respective trials, or clarify in our tutorials that we don't support the functionality of the old GUI, including loading the `S1_ongoing.txt` and `100_trials.txt` data files.

**Comment by jasmainak** [comment]
Agreed. If we separate the files into individual trials, it will break the old-GUI tutorials. Are you proposing to create files explicitly for the GUI in `hnn-core` ? Probably path of least resistance?

**Comment by rythorpe** [comment]
My vote would be to just clarify that we don't currently support reading in those two files so that we don't have to create a whole new data directory for .txt files that are solely compatible with hnn-core. Eventually we'll want to convert all our data files over to whatever file format we'll be supporting in the future (e.g., HDF5), but this seems premature at the moment. Maybe we can modify the gamma tutorial to reflect only trial-by-trial comparison in the frequency domain and encourage users to use hnn-core scripting to conduct more complex analysis workflows (e.g., comparison between average PSD between simulation trials and experimental trials)?

**Comment by chenghuzi** [comment]
> My vote would be to just clarify that we don't currently support reading in those two files so that we don't have to create a whole new data directory for .txt files that are solely compatible with hnn-core. Eventually we'll want to convert all our data files over to whatever file format we'll be supporting in the future (e.g., HDF5), but this seems premature at the moment. Maybe we can modify the gamma tutorial to reflect only trial-by-trial comparison in the frequency domain and encourage users to use hnn-core scripting to conduct more complex analysis workflows (e.g., comparison between average PSD between simulation trials and experimental trials)?

Could we also add some guards inside the `read_dipole`?
